src/Goerli-DssSpell.t.sol

```diff
@@ -458,4 +508,494 @@ contract DssSpellTest is GoerliDssSpellTestBase {
+        assertEq(dai.balanceOf(address(rwaconduitin_008)), 0, "RWA008: Dangling Dai in input conduit after push()");
+
+        rwaurn_008.wipe(daiToPay);
+        rwaurn_008.free(1 * WAD);
+
+        (ink, art) = vat.urns("RWA008-A", address(rwaurn_008));
+        assertEq(ink, preInk, "RWA008: Bad `ink` after free()");
+        assertLt(art, 4, "RWA008: Bad `art` - larger than conversion error dust after wipe()"); // wad -> rad conversion in wipe leaves some dust
+    }
+
+    function testRWA008_OPERATOR_LOCK_DRAW_CAGE() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        // set the balance of this contract
+        hevm.store(address(rwagem_008), keccak256(abi.encode(address(this), uint256(3))), bytes32(uint256(1 * WAD)));
+        assertEq(rwagem_008.balanceOf(address(this)), 1 * WAD, "RWA008: wrong gem balance");
+        // setting address(this) as operator
+        hevm.store(address(rwaurn_008), keccak256(abi.encode(address(this), uint256(1))), bytes32(uint256(1)));
+        assertEq(rwaurn_008.can(address(this)), 1, "RWA008: address(this) not operator");
+
+        rwagem_008.approve(address(rwaurn_008), 1 * WAD);
+        rwaurn_008.lock(1 * WAD);
+        assertEq(dai.balanceOf(address(rwaconduitout_008)), 0, "RWA008: dai balance is not zero for output conduit");
+        rwaurn_008.draw(1_000_000 * WAD);
+
+        (uint256 ink, uint256 art) = vat.urns("RWA008-A", address(rwaurn_008));
+        assertEq(ink, 1 * WAD, "RWA008: wrong ink in urn");
+        assertEq(art, 1_000_000 * WAD, "RWA008: wrong art in urn"); // rate == 1 RAY (drip was never called and passed 0 seconds after init)
+
+        // wards
+        giveAuth(address(rwaconduitout_008), address(this));
+        // can
+        hevm.store(address(rwaconduitout_008), keccak256(abi.encode(address(this), uint256(1))), bytes32(uint256(1)));
+        assertEq(rwaconduitout_008.can(address(this)), 1, "RWA008: output conduit not operator");
+        // may
+        hevm.store(address(rwaconduitout_008), keccak256(abi.encode(address(this), uint256(6))), bytes32(uint256(1)));
+        assertEq(rwaconduitout_008.may(address(this)), 1, "RWA008: output conduit not mate");
+
+        assertEq(dai.balanceOf(address(rwaconduitout_008)), 1_000_000 * WAD, "RWA008: wrong dai balance output conduit");
+
+        rwaconduitout_008.pick(address(this));
+
+        rwaconduitout_008.push();
+
+        assertEq(dai.balanceOf(address(this)), 1_000_000 * WAD, "RWA008: wrong dai balance address(this)");
+
+        giveAuth(address(end), address(this));
+        end.cage();
+        end.cage("RWA008-A");
+
+        end.skim("RWA008-A", address(rwaurn_008));
+
+        (ink, art) = vat.urns("RWA008-A", address(rwaurn_008));
+        assertEq(ink, 1 * WAD - (1_000_000 * WAD * WAD / 30_043_520_665599336150000000), "RWA008: wrong ink in urn after skim");
```

Comment:
This expression is a bit confusing at first look:

```
1 * WAD - (1_000_000 * WAD * WAD / 30_043_520_665599336150000000)
```

If I got this straight, `end.skim()` will take only enough worth of `gem` to cover the current outstanding debt of the vault.

The expression is **mathematically** equivalent of:

```solidity
1 * WAD - (1_000_000 * WAD) / (30_043_520_665599336150000000 / WAD)
```

However you make sure to multiply before divide to avoid rounding issues with Solidity.

It's a bit hard to visualize how we are making the units align though and the `* WAD * WAD` expression makes it confusing:

```solidity
gem*WAD        dai*WAD      /       (dai/gem)*WAD            / WAD
   v              v                       v                     v
1 * WAD - (1_000_000 * WAD) / (30_043_520_665599336150000000 / WAD)
```

If we try to simplify the expression on the right:
```
dai*WAD / (dai/gem)*WAD / WAD
dai*WAD / (dai/gem)
gem * WAD
```
then we get to the desired unit.

It took me some time to figure out what is going on here, so maybe we should document this step in the test for clarity:

```suggestion
        // `skim()` should take only enough gem value in DAI to cover the current debt position:
        //
        //    skimmedInk = currentDebt / gemPriceInDai
        //
        // The remaining `ink` can be obtained by:
        //
        //    ink = previousInk - skimmedInk
        //
        // We need to be careful because the gem price is stored as a WAD, so to make sure the units align in the
        // expression above, we need to divide the price by WAD.
        //
        //     gem*WAD       dai*WAD      /       (dai/gem)*WAD            / WAD
        //        v             v                       v                     v
        //    1 * WAD - (1_000_000 * WAD) / (30_043_520_665599336150000000 / WAD)
        //
        // While mathematically correct, the expression will present a rounding error with Solidity's integer math, so
        // we reorder it as follows:
        uint256 skimmedGem = (1_000_000 * WAD * WAD / 30_043_520_665599336150000000);
        assertEq(ink, 1 * WAD - skimmedGem, "RWA008: wrong ink in urn after skim");
```

Reply:
Also we should probably extract `30_043_520_665599336150000000` into a properly named constant:

```solidity
uint256 internal constant RWA008_A_PIP_VALUE = 30_043_520_665599336150000000;
```

Reply:
Personally I would have felt it was a bit too much explanation just for an assert in a test, however I agree it can be confusing and taking into account you have already done a great job explaining the case, no reason to not merge it.